src/neml2/tensors/LabeledMatrix.cxx

```diff
@@ -37,12 +37,25 @@ LabeledMatrix::identity(TensorShapeRef batch_size,
 }
 
 void
-LabeledMatrix::fill(const LabeledMatrix & other, bool recursive)
+LabeledMatrix::fill(const LabeledMatrix & other, bool common_first, bool recursive)
```

Comment:
I would write this function in a slightly different way

```cpp
void
LabeledMatrix::fill(const LabeledMatrix & other, Size odim, bool recursive)
{
  for (Size i = 0; i < axes().size() && i != odim; i++)
    neml_assert_dbg(axis(i) == other.axis(i), "Can only accumulate matrices with conformal common axes");
  const auto cindices = axis(odim).common_indices(other.axis(odim), recursive);
  for (const auto & [idx, idx_other] : cindices)
  {
    TensorIndices indices(base_dim(), indexing::Slice());
    TensorIndices indices_other(base_dim(), indexing::Slice());
    indices[odim] = idx;
    indices_other[odim] = idx_other;
    _tensor.base_index_put_(indices, other.base_index(indices_other));
  }
}
```

So that later on we could easily generalize this method to `LabeledVector` and `LabeledTensor3D` if we want, i.e., move it down to the `LabeledTensor` base class.

Reply:
I was going to do that, but it doesn't really make sense to allow for an arbitrary index for a matrix.  I guess we can provide this at the superclass and specialize.

Reply:
you are right, I agree